ui/qt-app/Glue.hs

```diff
@@ -39,347 +24,23 @@ import Ariadne.UX.PasswordManager
-
-    exprProcCall = Knit.ExprProcCall Knit.NoExt
-    exprLit = Knit.ExprLit Knit.NoExt
-    procCall = Knit.ProcCall Knit.NoExt
-    argPos = Knit.ArgPos Knit.NoExt
-    argKw = Knit.ArgKw Knit.NoExt
-
-commandIdToUI :: Unique -> Maybe TaskId -> UiCommandId
-commandIdToUI u mi =
-  UiCommandId
-    { cmdIdEqObject = fromIntegral (hashUnique u)
-    , cmdTaskIdRendered = fmap (\(TaskId i) -> show i) mi
-    , cmdTaskId = fmap (\(TaskId i) -> i) mi
-    }
-
--- The 'Maybe' here is not used for now, but in the future might be, if some
--- event couldn't be mapped to a UI event.
-knitCommandResultToUI
-  :: forall components.
-     ( AllConstrained Knit.ComponentPrinter components
-     , AllConstrained (Knit.ComponentInflate components) components
-     )
-  => UiCommandId
-  -> KnitCommandResult components
-  -> Maybe UiEvent
-knitCommandResultToUI commandId = Just . UiCommandEvent commandId . \case
-  KnitCommandSuccess v ->
-    UiCommandSuccess $ Knit.ppValue v
-  KnitCommandEvalError e ->
-    UiCommandFailure $ Knit.ppEvalError e
-  KnitCommandProcError e ->
-    UiCommandFailure $ Knit.ppResolveErrors e
-  KnitCommandException e ->
-    UiCommandFailure $ PP.text (displayException e)
-
-knitCommandOutputToUI :: UiCommandId -> PP.Doc -> UiEvent
-knitCommandOutputToUI commandId doc = UiCommandEvent commandId (UiCommandOutput doc)
-
-----------------------------------------------------------------------------
--- Glue between the Cardano backend and Qt frontend
-----------------------------------------------------------------------------
-
--- The 'Maybe' here is not used for now, but in the future might be, if some
--- event couldn't be mapped to a UI event.
-cardanoEventToUI :: CardanoEvent -> Maybe UiEvent
-cardanoEventToUI = \case
-  CardanoLogEvent message ->
-    Just $ UiBackendEvent $
-      UiBackendLogEvent message
-  CardanoStatusUpdateEvent CardanoStatusUpdate{..} ->
-    Just $ UiBackendEvent $
-      UiBackendStatusUpdateEvent UiBackendStatusUpdate
-        { syncProgress = (<> "%") . toFixed 1 . fromRational . (* 100) <$> syncProgress
-        , blockchainLocal = "block " <> pretty tipHeaderHash <> ", " <> pEpochOrSlot tipEpochOrSlot
-        , blockchainNetwork = pSlotId currentSlot
-        }
-  where
-    pEpoch = ("epoch " <>) . pretty . getEpochIndex
-    pSlotId SlotId{..} = pEpoch siEpoch <> ", slot " <> pretty (getSlotIndex siSlot)
-    pEpochOrSlot = either pEpoch pSlotId . unEpochOrSlot
-
-putCardanoEventToUI :: UiFace -> CardanoEvent -> IO ()
-putCardanoEventToUI UiFace{..} ev =
-  whenJust (cardanoEventToUI ev) putUiEvent
+putCommonUiEvent :: QtFace -> UiEvent -> IO ()
+putCommonUiEvent QtFace{..} event = putUiEvent $ UiCommonEvent event
 
 -- Called when an exception in backend thread occurs
-putBackendErrorToUI :: UiFace -> SomeException -> IO ()
-putBackendErrorToUI UiFace{..} e = putUiEvent . UiBackendExceptionEvent $
+-- TODO: Why is this not a thing in Vty?
```

Comment:
If backend fails in TUI, it's ok to have the program crash with a message printed to console (which is the default exception handling mechanism). In the GUI case, however, we want to show a dialog with the error. See #358.